crashlogs.cpp

```diff
@@ -182,12 +237,18 @@ namespace glaiel::crashlogs {
     void begin_monitoring() {
         output_thread = std::thread(crash_handler_thread);
 
+#ifdef OS_WINDOWS
         SetUnhandledExceptionFilter(exception_handler);
-        std::signal(SIGABRT, signal_handler);
-        std::set_terminate(terminator);
         _set_abort_behavior(0, _WRITE_ABORT_MSG | _CALL_REPORTFAULT);
         _set_purecall_handler(terminator);
         _set_invalid_parameter_handler(&invalid_parameter_handler);
+#endif
+        std::signal(SIGABRT, signal_handler);
+
+        std::signal(SIGSEGV, signal_handler);
+        std::signal(SIGILL, signal_handler);
+
+        std::set_terminate(terminator);

```

Comment:
prefer 1 version of begin_monitoring() per platform even if it means duplicated code vs chopping pieces of it out in the middle, these could end up diverging significantly and its a bit easier to read 

```
#ifdef OS_WINDOWS 
    void begin_monitoring() {
    ...
    }
#else
    void begin_monitoring() {
    ...
    }
#endif
```

Reply:
done
I also went ahead and restores your exact implementation for the windows version